### ISSUE TITLE
Simplifier does not work as expected

### ISSUE THREAD

**Issue by egecetintas-pointr** [Permafacture/Py-Visvalingam-Whyatt#9]
`polygon = [(90,100), (100,100), (100,150),(150,150), (150,100), (100,100), (90,100)]`
`simplified = vw.simplify(polygon,threshold=10)` 

For the given polygon, I'd expect the vertex (90,100) to be removed but the simplifier cannot handle that. I think there might be an issue with the implementation of the Visvalingam algorithm.

**Comment by Permafacture** [comment]
Yeah you're right. Looks like the endpoints aren't handled correctly. The implementation also doesn't specify if the start needs to be repeated at the end. It should expect it one way or the other (if start is repeated at end) and correct the input if it's given the other way. Then it should also handle the endpoints correctly, and there should be tests for if the first point, second point, and last point that isn't the repeated first point are the smallest.

I wrote this a LONG time ago and the test that's there sucks.

I'd be open to a pull request if you want to do it
